Topic 6: Spring Core & REST

hherminioandres opened the pull request:
1- Expose the shopping cart created in the topic 3.
2- Write the swagger file for point 1.
3- Create and document with swagger a REST API to register users. The API must provide: add, delete, update and find by name and find by nickname operations. Note that the username must be unique.

hherminioandres committed:
Topic 6: Spring Core & REST

1- Expose the shopping cart created in the topic 3.
2- Write the swagger file for point 1.
3- Create and document with swagger a REST API to register users. The API must provide: add, delete, update and find by name and find by nickname operations. Note that the username must be unique.

hherminioandres committed:
Unnecessary libraries deleted

sebastian-garofalo commented on topic_6/src/main/java/com/globant/shopping/ShoppingCart.java:
```diff
@@ -0,0 +1,81 @@
+ * @author Herminio Andres Hernandez
+ * @version 21/07/2018
+ */
+public class ShoppingCart {
```
Why a shopping cart without product quantity?

sebastian-garofalo commented on topic_6/src/main/java/com/globant/shopping/controller/ShopingCartController.java:
```diff
@@ -0,0 +1,41 @@
+	public float calculate() {
+		return shoppingCart.calculate();
+	}
+    @RequestMapping("/addProduct")
```
Why are you always using the GET method, what about POST, DELETE and the rest of the HTTP verbs?

sebastian-garofalo commented on topic_6/src/main/java/com/globant/user/User.java:
```diff
@@ -0,0 +1,66 @@
+		return user;
+	}
+
+	public void setNickname(String userName) {
```
The setter should use the same name as the variable it sets to avoid confussions

sebastian-garofalo commented on topic_6/src/main/java/com/globant/user/User.java:
```diff
@@ -0,0 +1,66 @@
+		return firstname;
+	}
+
+	public void setFirtname(String firstname) {
```
there is a typo in the method name

sebastian-garofalo commented on topic_6/src/main/java/com/globant/user/controller/UserRepositoryController.java:
```diff
@@ -0,0 +1,49 @@
+
+	UserRepository userRepository = new UserRepository();
+
+	@RequestMapping("/createUser")
```
Too much GET and no other methods, REST core principle is that HTTP methods represent the different possible operations.
You used them in the swagger files but not in the code, you can do something like
@RequestMapping(value = "/ex/foos", method = POST)

or

use the annotations like

GetMapping, PutMapping, DeleteMapping, PatchMapping, RequestMapping

hherminioandres committed:
Topic_6: Several corrections

- Added productQuantity method  in ShoppingCart class
- Applied methods GET, SET, DELETE, POST and PUT
- Typo corrections

hherminioandres committed:
Tab correction

hherminioandres committed:
error
